lib/bindings/MQTTBinding.js

```diff
@@ -236,15 +253,21 @@ function start(callback) {
     const retries = mqttConfig.retries ? mqttConfig.retries : constants.MQTT_DEFAULT_RETRIES;
     const retryTime = mqttConfig.retryTime ? mqttConfig.retryTime : constants.MQTT_DEFAULT_RETRY_TIME;
     let isConnecting = false;
-    let numRetried = 0;
-    config.getLogger().info(context, 'Starting MQTT binding');
+    let numRetried = 1; // retries will be disabled when MQTT_DEFAULT_RETRIES=0
+    config
+        .getLogger()
+        .debug(context, 'Starting MQTT binding with options %j retries %s retryTIme %s', options, retries, retryTime);
 
     function createConnection(callback) {
-        config.getLogger().info(context, 'creating connection');
+        config.getLogger().debug(context, 'creating connection');
         if (isConnecting) {
             return;
         }
         isConnecting = true;
+        // Ensure clientId is unique when reconnect to avoid loop closing old connection which the same name
+        options.clientId = mqttConfig.clientId
+            ? mqttConfig.clientId
```

Comment:
Maybe mqttConfig.clientId should be mqttConfig.clientId + Math.random().toString(16).substr(2, 8)